transport.go

```diff
@@ -17,75 +17,118 @@ const LegacyTransportID TransportID = "graphsync"
 // i.e. graphsync 1.0.0
 var LegacyTransportVersion Version = Version{1, 0, 0}
 
+type TransportEvent interface {
+	transportEvent()
+}
+
+// TransportOpenedChannel occurs when the transport begins processing the
+// request (prior to that it may simply be queued) -- only applies to initiator
+type TransportOpenedChannel struct{}
+
+// TransportInitiatedTransfer occurs when the transport actually begins sending/receiving data
+type TransportInitiatedTransfer struct{}
+
+// TransportReceivedData occurs when we receive data for the given channel ID
+// index is a transport dependent of serializing "here's where I am in this transport"
+type TransportReceivedData struct {
+	Size  uint64
+	Index datamodel.Node
+}
+
+// TransportSentData occurs when we send data for the given channel ID
+// index is a transport dependent of serializing "here's where I am in this transport"
+type TransportSentData struct {
+	Size  uint64
+	Index datamodel.Node
+}
+
+// TransportQueuedData occurs when data is queued for sending for the given channel ID
+// index is a transport dependent of serializing "here's where I am in this transport"
+type TransportQueuedData struct {
+	Size  uint64
+	Index datamodel.Node
+}
+
+// TransportReachedDataLimit occurs when a channel hits a previously set data limit
+type TransportReachedDataLimit struct{}
+
+/*
+type TransportReceivedVoucherRequest struct {
+	Request Request
+}
+
+type TransportReceivedUpdateRequest struct {
+	Request Request
+}
+
+type TransportReceivedCancelRequest struct {
+	Request Request
+}
+
+// TransportReceivedResponse occurs when we receive a response to a request
+type TransportReceivedResponse struct {
+	Response Response
+}
```

Comment:
what's the story with these? are they are TODO in order to finish this work? they seem like important pieces of the puzzle

Reply:
ah, I think these come under:

> there are some legacy "Transport Events" from experiments with making more stuff asynchronous commented out

But I'd like to hear more about these specific events - were they never actually used? Are we losing anything by not having them?

Reply:
so my original plan, which I'd love your feedback on, was to make the whole thing an event stream, which would remove the requirement of any synchronous hooks. As it turns out, that's kinda hard. This was an in between point when I thought may be I could make this all work for just incoming new or restart requests, which I may still consider. But, honestly, this was the first final interface I could get stood up and running after trying several times!